contracts/erc20/erc20_capped.aes

```diff
@@ -0,0 +1,103 @@
+contract ERC20Capped =
+  record state = {
+    _owner            : address,
+    _cap              : int,
+    _totalSupply      : int,
+    _balances         : map(address, int),
+    _allowed          : map((address,address), int)}
+
+  public stateful function init(cap : int) : state = 
+    require(cap > 0, "Cap can't be less than a zero")
+    { _owner       = Call.caller,
+      _cap         = cap,
+      _totalSupply = 0,
+      _balances    = {},
+      _allowed     = {}}
+
+  private function lookupByAddress(k : address, m, v) =
+  	switch(Map.lookup(k, m))
+	    None    => v
+	    Some(x) => x
+
+  public function totalSupply() : int = state._totalSupply
+
+  public function balanceOf(who: address) : int = lookupByAddress(who, state._balances, 0)
+
+  public function allowance(owner: address, spender: address) : int = 
+    switch(Map.lookup((owner, spender), state._allowed))
+	    None    => 0
+	    Some(x) => x
+
+  public stateful function transfer(to: address, value: int) : bool =
+    _transfer(Call.caller, to, value)
+
+  public stateful function approve(spender: address, value: int) : bool = 
+    require(value > 0, "Value is sub zero")
+    require(spender != #0, "Invalid spender address")
+
+    put(state{_allowed[(Call.caller,spender)] = value})
+
+    true
+
+  private stateful function _transfer(from: address, to: address, value: int) : bool =
+    require(value > 0, "Value is sub zero")
+    require((value =< balanceOf(from)), "Not enough balance")
+    require(to != #0, "Invalid address")
+
+    put(state{
+      _balances[from] = sub(state._balances[from], value),
+      _balances[to] = add(state._balances[to], value)})
+
+    true
+
+  public stateful function transferFrom(from: address, to: address, value: int) : bool =
```

Comment:
Same as the comment from ERC20, if true change in all occurrences

Reply:
-||-